### PR TITLE
T295964 Custom story title

### PR DESCRIPTION
**Pull request by hueitan**
Phabricator: https://phabricator.wikimedia.org/T295964
Demo: https://wikimedia.github.io/wikistories-prototype/T295964-edit-title/#/

 - [x] Publish Page Edit Form
 - [x] View Cover page on Story Viewer
 - [x] i18n
 - [x] Style Polish
 - [x] Code Review Update

| Publish Page Error | Scrollable Edit Area | Cover Page | Truncate the Cover Title |
| --- | --- | --- | --- |
| image | image | image | image |

### CODE REVIEW

File: src/i18n/en.js [hunk 1]
```diff
@@ -12,6 +12,8 @@ export default {
   'btn-clear': 'Clear',
   'btn-select-image': 'Select an image',
   'btn-next': 'Next',
+  'publish-edit-title-placeholder': 'Give your story a title',
+  'publish-edit-error-msg': 'Enter story title',
```
Comment (hueitan) [inline src/i18n/en.js:16]: @medied @stephanebisson When you have time can you add the other translation?

File: src/views/Publish.vue [hunk 2]
```diff
@@ -1,9 +1,10 @@
 <template>
     <div class="view publish">
         <Navigator :onBack="() => this.$router.push( { name: 'Story' } )" />
-        <div class="header">
-            <h2 class="title" v-html="storyInfo.title"></h2>
+        <div :class="`header ${isFormError?'error':''}`">
```
Comment (stephanebisson) [inline src/views/Publish.vue:4]: Did you know that `:class` also accept an object like `{ class: <condition> }`. For example: `:class="{ header: true, error: isFormError }"`

File: src/i18n/en.js (hunk 1 above)
Comment (stephanebisson) [reply src/i18n/en.js:16]: Here's the french version: 
  'publish-edit-title-placeholder': 'Donnez un title à votre histoire',
  'publish-edit-error-msg': 'Veuillez entrer un title pour votre histoire',

Comment (hueitan) [reply src/i18n/en.js:16]: I'm waiting for @medied then add them together.

Comment (medied) [reply src/i18n/en.js:16]: Portuguese:

```
 'publish-edit-title-placeholder': 'Dê um título à sua história',
 'publish-edit-error-msg': 'Insira o título da história',
```

Spanish:
```
 'publish-edit-title-placeholder': 'Ponle un titulo a tu historia',
 'publish-edit-error-msg': 'Agregar titulo',
```

**Review by medied — APPROVED** [review]
LGTM

File: src/store/story.js [hunk 3]
```diff
@@ -16,10 +16,16 @@ const makeFrameStyle = f => {
 
 export default {
   state: {
-    storyTitle: '', // @todo to be edited by user
     creationDate: null,
     currentFrameId: 1,
     frames: [
+      { // cover frame
```
Comment (stephanebisson) [inline src/store/story.js:22]: By adding the cover in the official list of frame, a lot of unrelated code is force to filter it out to still function properly. Would it be simpler to add it only in a getter used specifically by the viewer? The story title can be stored as a top-level field in the state.

Comment (hueitan) [reply src/store/story.js:22]: Initially I was planning to do that, therefore `storyTitle` existed, with this flow the `currentFrame` and `currentFrameId` will need to be updated according too.

I agree with the `filter( f => f.id === 0 )` is really unrelated and not clear, I have one idea to try out, let me do the update later here.

Comment (hueitan) [reply src/store/story.js]: Done, please test this carefully once again.

**Comment by medied** [comment]
I like that last commit, testing well for me

File: src/store/story.js [hunk 4]
```diff
@@ -114,20 +120,22 @@ export default {
...
+      const f = isCoverFrame ? state.frames[0] : state.frames.find(f => f.id === state.currentFrameId)
       return {
-        text: f.text,
+        text: isCoverFrame ? state.storyTitle : f.text,
         style: makeFrameStyle(f),
         noImage: f.img === '',
         id: state.currentFrameId,
         imgAttribution: f.attribution,
-        imgTitle: f.imgTitle
+        imgTitle: isCoverFrame ? state.storyTitle : f.imgTitle
```
Comment (stephanebisson) [inline src/store/story.js:131]: I don't think this property is used at all, I can't find it.

It should be removed in a different PR.

**Review by stephanebisson — APPROVED** [review]
